Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/SettingsRegistrar.h

```diff
@@ -0,0 +1,90 @@
+        //! Specifying a json pointer path prefix in the DumperSettings may be required in order
+        //! for the dumped json data to match the in-memory path structure.
+        //! The path passed to this function is expected to be relative to the project root.
+        AZ::Outcome<void, AZStd::string> SaveSettingsToFile(
+            const AZStd::string& relativeFilepath,
+            AZ::SettingsRegistryMergeUtils::DumperSettings dumperSettings,
+            const AZStd::string& rootSearchKey = "") const;
+
+        //! Loads settings from the provided '.setreg' file and merges settings into the SettingsRegistry
+        //! at the given anchor key.
+        //! The path passed to this function is expected to be relative to the project root.
+        AZ::Outcome<void, AZStd::string> LoadSettingsFromFile(
+            AZStd::string_view relativeFilepath,
+            AZStd::string_view anchorKey = "",
+            AZ::SettingsRegistryInterface::Format format = AZ::SettingsRegistryInterface::Format::JsonMergePatch) const;
+
+        //! Attempts to retrieve settings stored at the given registry path and put them in the provided object.
+        //! The provided object is expected to be intialized before being passed to this function and it must
+        //! be AZ_RTTI enabled.
+        template<typename AzRttiEnabled_T>
+        AZ::Outcome<void, AZStd::string> GetObjectSettings(AzRttiEnabled_T* outSettingsObject, AZStd::string_view registryPath) const
+        {
+            AZ::SettingsRegistryInterface* registry = AZ::SettingsRegistry::Get();
+            if (!registry)
+            {
+                return AZ::Failure(AZStd::string::format("Failed to access global settings registry for data retrieval"));
+            }
+
+            if (registry->GetObject(*outSettingsObject, registryPath))
+            {
+                return AZ::Success();
+            }
+            else
+            {
+                return AZ::Failure(AZStd::string::format("Failed to retrieve settings at path '%s'", registryPath.data()));
+            }
+        }
+
+        //! Attempts to store reflected properties of the given AZ_RTTI enabled object at the given registry path.
+        template<typename AzRttiEnabled_T>
+        AZ::Outcome<void, AZStd::string> StoreObjectSettings(AZStd::string_view registryPath, AzRttiEnabled_T* settingsObject) const
+        {
+            AZ::SettingsRegistryInterface* registry = AZ::SettingsRegistry::Get();
+            if (!registry)
+            {
+                return AZ::Failure(AZStd::string::format("Failed to access global settings registry for data storage"));
+            }
+
+            if (registry->SetObject(registryPath, *settingsObject))
+            {
+                return AZ::Success();
+            }
+            else
+            {
+                return AZ::Failure(AZStd::string::format("Failed to store settings at path '%s'", registryPath.data()));
+            }
+        }
```

Comment:
I recommend the functions within this file have an overload that accepts a `SettingsRegistryInterface*` or add have a default parameter of `SettingsRegistryInterface* = nullptr`

That way the user can supply their own Settings Registry if they don't want to use the global SettingsRegistry.

If the settings registry pointer is nullptr, then the global registry is used as a fallback.

Reply:
Thanks. Added a default `SettingsRegistryInterface*` param to the functions in this file which is set to the global registry when nullptr.